Using C++23 builds with warning: `std::aligned_storage` is deprecated

In C++23, `std::aligned_storage` was deprecated with [P1413R3](https://wg21.link/P1413R3).

Using it leads to a compiler warning with MSVC:

```
api\include\opentelemetry\nostd\internal\absl\meta\type_traits.h(117,50): warning C4996: 'std::aligned_storage<_Len,_Align>': warning STL4034: std::aligned_storage and std::aligned_storage_t are deprecated in C++23. Prefer alignas(T) std::byte t_buff[sizeof(T)]. You can define _SILENCE_CXX23_ALIGNED_STORAGE_DEPRECATION_WARNING or _SILENCE_ALL_CXX23_DEPRECATION_WARNINGS to suppress this warning.
```

While `std::aligned_storage` isn't actually used for anything in `opentelemetry-cpp` or its dependencies, it _is_ referenced in that one file: [`api/include/opentelemetry/nostd/internal/absl/meta/type_traits.h`](https://github.com/open-telemetry/opentelemetry-cpp/blob/994761988eed36a41014de6291d470c738190ae0/api/include/opentelemetry/nostd/internal/absl/meta/type_traits.h#L624).

That file uses it to define the no-std version of `aligned_storage_t`, which doesn't appear to be used anywhere.

### Proposal

Remove `aligned_storage` from `nostd`.
This is exactly what Abseil did as well, in https://github.com/abseil/abseil-cpp/commit/42133464db0ae5383fef723067835ef768b849f8.

--- a/api/include/opentelemetry/nostd/internal/absl/meta/type_traits.h
+++ b/api/include/opentelemetry/nostd/internal/absl/meta/type_traits.h
@@ -106,18 +106,6 @@ struct VoidTImpl {
   using type = void;
 };
 
-// This trick to retrieve a default alignment is necessary for our
-// implementation of aligned_storage_t to be consistent with any implementation
-// of std::aligned_storage.
-template <size_t Len, typename T = std::aligned_storage<Len>>
-struct default_alignment_of_aligned_storage;
-
-template <size_t Len, size_t Align>
-struct default_alignment_of_aligned_storage<Len,
-                                            std::aligned_storage<Len, Align>> {
-  static constexpr size_t value = Align;
-};
-
 ////////////////////////////////
 // Library Fundamentals V2 TS //
 ////////////////////////////////
@@ -618,10 +606,6 @@ using remove_extent_t = typename std::remove_extent<T>::type;
 
 template <typename T>
 using remove_all_extents_t = typename std::remove_all_extents<T>::type;
-
-template <size_t Len, size_t Align = type_traits_internal::
-                          default_alignment_of_aligned_storage<Len>::value>
-using aligned_storage_t = typename std::aligned_storage<Len, Align>::type;
 
 template <typename T>
 using decay_t = typename std::decay<T>::type;

